### ISSUE TITLE
Reset equation numbering after each file or page

### ISSUE THREAD

**Issue by najuzilu** [ExecutableBookProject/cli#68]
It might be helpful to include something like this in the toc file. I don't see a way to do that now but I might be wrong.

**Comment by choldgraf** [comment]
hmmm - I am not sure how this usually works in Sphinx, but it seems reasonable to allow people to reset equation numbers. I think maybe it's controllable with the `{math}` directive?

**Comment by choldgraf** [comment]
@najuzilu is this one still an issue for you? I'm not sure it's something that we could fix at the CLI level, but correct me if I'm wrong

**Comment by najuzilu** [comment]
I think the solution would be to use `:math_numfig:` with `:numfig_secnum_depth:`. This is generally added at the `toctree` directive in Sphinx. I'm guessing we don't currently support this at the CLI level, correct?

**Comment by choldgraf** [comment]
nope, the CLI doesn't trigger behavior like that, but users could configure it in their directives or in their configuration files as needed

**Comment by jstac** [comment]
I'm not sure exactly what's being requested but what would be really perfect here is if the equation numbers matched across the PDF and HTML. 

@boazbk has implemented this (at least for theorems) in https://github.com/boazbk/tcs.  

Either way, my vote is that eq numbers are universal rather than local, so "equation 3.2" always refers to exactly one equation in the book.

**Comment by choldgraf** [comment]
I think "equation numbers are universal" should definitely be the default as well

**Comment by boazbk** [comment]
The  way I implement all references in my book is that I run latex first, and then have a python script that scans the aux file to get the numbers that correspond to all references.  See https://github.com/boazbk/panbook/blob/master/auxtoyaml.py and `getlabel` in https://github.com/boazbk/panbook/blob/master/book-filter.py

**Comment by najuzilu** [comment]
@jstac Our lectures' eq counter - see [scalar_dynam](https://python-intro.quantecon.org/scalar_dynam.html) and [finite_markov](https://python-intro.quantecon.org/finite_markov.html)- is reset to 1 for every lecture. This seems to be the default behavior. Currently, `quantecon-example` does not reflect this.

@choldgraf I'll see if this can be done at the directive level and report back.

Something else to consider is adding [numfig_format](https://www.sphinx-doc.org/en/master/usage/configuration.html). It would be helpful to format eq/table/fig labels at the toc level. What do others think?

@jstac this is similar to #92 but set at the toc level instead of manually updating the label.

**Comment by jstac** [comment]
Thanks @najuzilu.  I think there's agreement now that equation numbers should not reset at each lecture, and should be unique through the document.  Ideally, they would take the LaTeX form (3.2) for second equation of third lecture, etc.

Even better if they can match the PDF numbers, although that might be a long run target.

> It would be helpful to format eq/table/fig labels at the toc level. 

I'm not sure what this means.  Can you please give an example?

**Comment by najuzilu** [comment]
@jstac thanks for clarifying. 

> I'm not sure what this means. Can you please give an example?

The `numfig_format` is used as a caption prefix for referencing figures/tables etc.. The default behavior is "Fig. 1 <_caption_>". It allows the user to reset the prefix caption to something else, for example "Figure x: <_caption_>", but this is done at the toc level for all figures.

**Comment by choldgraf** [comment]
When you say toc level, do you mean _config.yml? That's the bit I'm confused about

**Comment by najuzilu** [comment]
Yes, apologies for the confusion. It would have to be under _config.yml

**Comment by choldgraf** [comment]
ah - in that case I agree, I think it's a good idea :-) 

here's the config section where we'd control this, so the question is just where in the `_config.yml` file we want that configuration to go: https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-numfig_format
